examples/kops-with-vpc/kubernetes/Makefile

```diff
@@ -0,0 +1,96 @@
+.PHONY: kops-create-cluster kops-edit-cluster kops-update-cluster \
+	kops-validate-cluster create-s3-bucket generate-ssh-keypair fetch-keys \
+	kops-rolling-update
+
+.DEFAULT_GOAL = help
+
+export PASSWORD_STORE_DIR ?= ${PWD}/../password-store
+
+require-%:
+	@ if [ "${${*}}" = "" ]; then \
+		echo "ERROR: Environment variable not set: \"$*\""; \
+		exit 1; \
+	fi
+
+## Fetch private/public ssh keys from pw store
+fetch-keys: require-ENVIRONMENT require-CLUSTER_NAME
+	@pass clusters/${CLUSTER_NAME}/keys/admin.pem.gpg > ${ENVIRONMENT}/${CLUSTER_NAME}-admin.pem
+	@pass clusters/${CLUSTER_NAME}/keys/admin.pem.pub.gpg > ${ENVIRONMENT}/${CLUSTER_NAME}-admin.pem.pub
+
+## Create s3 bucket for kops state
+create-s3-bucket: require-ENVIRONMENT require-CLUSTER_NAME
```

Comment:
Should the Terraform create this bucket instead of making it a separate step?

Reply:
I was on the fence about this one too... Reason I kept it this way was to have dependencies separate between the directories. But for this specific example, it does make sense to have it done in Terraform and update the Kops `env` accordingly.